itkwidgets/_to_itk.py

```diff
@@ -47,10 +47,11 @@ def to_itk_image(other_image_datatype):
         image_from_array.SetSpacing(other_image_datatype.GetSpacing())
         image_from_array.SetOrigin(other_image_datatype.GetOrigin())
         return image_from_array
-    elif have_imglyb and isinstance(other_image_datatype,
-            imglyb.util.ReferenceGuardingRandomAccessibleInterval):
-        array = imglyb.to_numpy(other_image_datatype)
-        image_from_array = itk.GetImageViewFromArray(array)
-        return image_from_array
+    elif have_imagej:
+        import imglyb
+        if isinstance(other_image_datatype, imglyb.util.ReferenceGuardingRandomAccessibleInterval):
+            array = imglyb.to_numpy(other_image_datatype)
+            image_from_array = itk.GetImageViewFromArray(array)
+            return image_from_array
```

Comment:
In future, this logic could be generalized to something like:
```python
        elif have_imagej and ij.py.isjava(other_image_datatype):
            converted = ij.py.from_java(other_image_datatype)
            if ij.py.isjava(converted):
                # Cannot escape from Java land...
                return None
            return to_itk_image(converted)
```

Reply:
Nice! Thanks -- added a note to Issue #137

Reply:
I am a little surprised that the current code is working. In my tests just now, I am seeing:
```python
>>> import imglyb, numpy
>>> z = numpy.zeros([5, 7])
>>> jz = imglyb.to_imglib(z)
>>> jz
<net/imglib2/python/ReferenceGuardingRandomAccessibleInterval at 0x13915d1a8 jclass=net/imglib2/python/ReferenceGuardingRandomAccessibleInterval jself=<LocalRef obj=0x7fa92c470618 at 0x10d9a06f0>>
>>> imglyb.to_numpy(jz)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/miniconda3/envs/imagej/lib/python3.7/site-packages/imglyb/__init__.py", line 50, in to_numpy
    return _ImgLibReferenceGuard(source)
  File "/usr/local/miniconda3/envs/imagej/lib/python3.7/site-packages/imglyb/imglib_ndarray.py", line 58, in __new__
    address = get_address(rai)
  File "/usr/local/miniconda3/envs/imagej/lib/python3.7/site-packages/imglyb/imglib_ndarray.py", line 50, in get_address
    raise ValueError("Excpected ArrayImg or UnsafeImg but got {}".format(class_name))
ValueError: Excpected ArrayImg or UnsafeImg but got ReferenceGuardingRandomAccessibleInterval
```

Reply:
Currently it is not possible to convert back to numpy from a `ReferenceGuardingRandomAccessibleInterval`. The only reason is that it has not been implemented, though.